### ISSUE TITLE
Another wrapper around jsmn

### ISSUE THREAD

**Issue by gllmflndn** [Andy1978/octave-rapidjson#1]
Hello,

I just came across your interesting project. I'm not opening an issue but just making contact!
I wrote a wrapper around a C library, [jsmn](https://github.com/zserge/jsmn) for MATLAB and Octave using the MEX interface:
- https://www.artefact.tk/software/matlab/jsonio/
- https://github.com/gllmflndn/JSONio

In the meantime, JSON support was added to MATLAB through [jsonencode](http://www.mathworks.com/help/matlab/ref/jsonencode.html)/[jsondecode](http://www.mathworks.com/help/matlab/ref/jsondecode.html) and I tried to be compatible with it.

I will be curious to see the speed you achieve with RapidJSON.

**Comment by Andy1978** [comment]
Hi, I made some tests with this JSON input https://github.com/Andy1978/octave-rapidjson/blob/devel/tests/big1.json and my rapidjson wrapper was 8.2 times faster than JSONio on GNU OCtave 4.3.0+

Btw, you really should include some steps to create the mex interface for GNU Octave in teh README if you want Octave users to use it.

The results is quite different: You're trying to create a cell array if the fieldnames match in adjacent structures, right? I don't have (and want) Matlab so I don't bother to be Matlab compatible here. I tried to be more compatible with jsonlab because this is what I've used before.

Happy hacking, Andy
